pcdscalc/be_lens_calcs.py

```diff
@@ -0,0 +1,826 @@
+"""Module for Beryllium Lens Calculations."""
+import logging
+import os
+import shutil
+from datetime import date
+from itertools import product
+
+import numpy as np
+import xraydb as xdb
+
+logger = logging.getLogger(__name__)
+
+# Set of Be lenses with thicknesses.
+LENS_RADII = [50e-6, 100e-6, 200e-6, 300e-6, 500e-6, 1000e-6, 1500e-6]
+# Constant for converting between FWHM and sigma of a Gaussian function.
+FWHM_SIGMA_CONVERSION = 2.35482004503
+# Constant for converting between wavelength and photon energy.
+WAVELENGTH_PHOTON = 1.2398
+
+
+def photon_to_wavelength(energy):
+    """
+    Find the wavelength in micrometers.
+
+    Use photon energy in electronvolts. The equation is approximately: λ[µm] =
+    1.2398 / E[eV]. The photon energy at 1 μm wavelength, the wavelength of
+    near infrared radiation, is approximately 1.2398 eV.
+
+    Usage:
+
+    .. code-block:: python
+
+        photon_to_wavelength(8)
+
+    Parameters
+    ----------
+    energy : number
+        Photon energy in electronvolts
+
+    Returns
+    -------
+    Wavelength in micrometers
+    """
+    return WAVELENGTH_PHOTON / energy
+
+
+def gaussian_sigma_to_fwhm(sigma):
+    """
+    Convert between FWHM and sigma of a Gaussian function.
+
+    FWHM = 2.35482004503 * sigma
+
+    https://brainder.org/2011/08/20/gaussian-kernels-convert-fwhm-to-sigma/
+
+    Usage:
+
+    .. code-block:: python
+
+        gaussian_sigma_to_fwhm(0.3)
+
+
+    Parameters
+    ----------
+    sigma : float
+
+    Returns
+    -------
+    FWHM -  Full Width at the Half Maximum
+    """
+    return FWHM_SIGMA_CONVERSION * sigma
+
+
+def gaussian_fwhm_to_sigma(fwhm):
+    """
+    Convert between FWHM and sigma of a Gaussian function.
+
+    sigma = FWHM / 2.35482004503
+
+    https://brainder.org/2011/08/20/gaussian-kernels-convert-fwhm-to-sigma/
+
+    Usage:
+
+    .. code-block:: python
+
+        gaussian_fwhm_to_sigma(0.023)
+
+    Parameters
+    ----------
+    fwhm : float
+        Full Width at the Half Maximum
+
+    Returns
+    -------
+    sigma
+    """
+    return fwhm / FWHM_SIGMA_CONVERSION
+
+
+# TODO: this is a test file... take care of it
```

Comment:
Well in the old code it seems like we had a fixed path for this as a default, so I don't know if we should still have that here or if we should leave it up to the user, but also, if we do need a default path for it, how should I go about it exaclty?

https://github.com/pcdshub/old-hutch-configs/blob/18c7769d66a647ab558ad7ebc34d019fc98fbf48/xpp/blutil/belenses.py#L15-L21

Reply:
We probably want a way to configure the path to the lens set file, and then we would set it up as needed in the hutch python loader. Then the default argument would be "whatever the user last configured".

The approach in the old hutch python (creating a dummy config module) should not be used, as this can cause package incompatibilities. I'd advocate for simple configuration functions in this module, e.g. `configure_lens_set_file` or some equivalent.

Reply:
Ok @ZLLentz  I like this idea, but I am not sure if I implemented as you intended in 
I added it in 5729619 but have a couple of fixes in 0244c55 and dd34461 - please let me know if this implementation if fine.

Reply:
Seems good to me